pages/MotdAdmin.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import Panel from '../Components/Site/Panel';
+import TextArea from '../Components/Form/TextArea';
+import RadioButton from '../Components/Form/RadioButton';
+import * as actions from '../actions';
+
+class MotdAdmin extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            motdText: props.motd && props.motd.message,
+            selectedMotdType: props.motd ? props.motd.motdType : 'info'
+        };
+
+        this.motdTypes = [
+            { value: 'error', label: 'Error (red)' },
+            { value: 'warning', label: 'Warning (yellow)' },
+            { value: 'info', label: 'Info (blue)' },
+            { value: 'success', label: 'Success (green)' }
+        ];
+
+        this.onMotdTextChange = this.onMotdTextChange.bind(this);
+        this.onSaveClick = this.onSaveClick.bind(this);
+    }
+
+    componentWillReceiveProps(props) {
+        this.setState({ motdText: props.motd && props.motd.message, selectedMotdType: props.motd ? props.motd.motdType : 'info' });
+    }
+
+    onMotdTextChange(event) {
+        this.setState({ motdText: event.target.value });
+    }
+
+    onMotdTypeChange(value) {
+        this.setState({ selectedMotdType: value });
+    }
+
+    onSaveClick(event) {
+        event.preventDefault();
+
+        this.props.sendSocketMessage('motd', { message: this.state.motdText, motdType: this.state.selectedMotdType });
+    }
+
+    render() {
+        let types = this.motdTypes.map(type => {
+            return (<RadioButton key={ type.value }
+                name={ type.value }
+                onChange={ this.onMotdTypeChange.bind(this, type.value) }
+                checked={ this.state.selectedMotdType === type.value }
```

Comment:
Minor nitpick, I think a stronger component design could take in the value associated with the radio button, and the currently selected value, and compare them internally, rather than controlling the `checked` property directly.

Even better if you could associate multiple radio buttons with the same "group" and store the selected value there (and potentially get rid of the onChange here), but offhand I'm not sure how you would structure the components that way.